File: csharp/my-widget-service/src/MyWidgetService/MyWidgetServiceStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.APIGateway;
using Amazon.CDK.AWS.Lambda;
using Amazon.CDK.AWS.S3;
using Constructs;
using System.Collections.Generic;

namespace MyWidgetService
{
    internal sealed class MyWidgetServiceStack : Stack
    {
        public MyWidgetServiceStack(Construct parent, string id, IStackProps props) : base(parent, id, props)
        {
            var bucket = new Bucket(this, "WidgetStore");

            var handler = new Function(this, "WidgetHandler", new FunctionProps {
                Runtime = Runtime.NODEJS_10_X,
                Code = Code.FromAsset("src/MyWidgetService/resources"),
                Handler = "widgets.main",
                Environment = new Dictionary<string, string>{
                    { "BUCKET", bucket.BucketName }
                }
            });

            bucket.GrantReadWrite(handler);

            var api = new RestApi(this, "widgets-api", new RestApiProps {
                RestApiName = "Widget Service",
                Description = "This service serves widgets"
            });

            var getWidgetsIntegration = new LambdaIntegration(handler, new LambdaIntegrationOptions{
                RequestTemplates = new Dictionary<string, string>{
                    { "application/json", "{ 'statusCode', '200'}" }
                },
            });

            api.Root.AddMethod("GET", getWidgetsIntegration);

            var widget = api.Root.AddResource("{id}");

            var postWidgetIntegration = new LambdaIntegration(handler);
            var getWidgetIntegration = new LambdaIntegration(handler);
            var deleteWidgetIntegration = new LambdaIntegration(handler);

            widget.AddMethod("POST", postWidgetIntegration);
            widget.AddMethod("GET", getWidgetIntegration);
            widget.AddMethod("DELETE", deleteWidgetIntegration);
        }
    }
}

File: java/my-widget-service/src/main/java/software/amazon/awscdk/examples/MyWidgetServiceStack.java
package software.amazon.awscdk.examples;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import software.constructs.Construct;
import software.amazon.awscdk.Duration;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.services.apigateway.LambdaIntegration;
import software.amazon.awscdk.services.apigateway.Resource;
import software.amazon.awscdk.services.apigateway.RestApi;
import software.amazon.awscdk.services.iam.IManagedPolicy;
import software.amazon.awscdk.services.iam.ManagedPolicy;
import software.amazon.awscdk.services.iam.Role;
import software.amazon.awscdk.services.iam.ServicePrincipal;
import software.amazon.awscdk.services.lambda.Code;
import software.amazon.awscdk.services.lambda.Function;
import software.amazon.awscdk.services.lambda.Runtime;
import software.amazon.awscdk.services.s3.Bucket;

public class MyWidgetServiceStack extends Stack {
  public MyWidgetServiceStack(final Construct scope, final String id) {
    super(scope, id, null);

    Bucket bucket = Bucket.Builder.create(this, "WidgetStore").build();

    RestApi api =
        RestApi.Builder.create(this, "widgets-api")
            .restApiName("Widget Service")
            .description("This service serves widgets.")
            .build();

    List<IManagedPolicy> managedPolicyArray = new ArrayList<IManagedPolicy>();
    managedPolicyArray.add(
        (IManagedPolicy) ManagedPolicy.fromAwsManagedPolicyName("AmazonS3FullAccess"));

    Role.Builder.create(this, "RestAPIRole")
        .assumedBy(new ServicePrincipal("apigateway.amazonaws.com"))
        .managedPolicies(managedPolicyArray)
        .build();

    Map<String, String> environmentVariables = new HashMap<String, String>();
    environmentVariables.put("BUCKET", bucket.getBucketName());

    Function lambdaFunction =
        Function.Builder.create(this, "WidgetHandler")
            .code(Code.fromAsset("resources"))
            .handler("widgets.main")
            .timeout(Duration.seconds(300))
            .runtime(Runtime.NODEJS_16_X)
            .environment(environmentVariables)
            .build();

    bucket.grantReadWrite(lambdaFunction);

    Map<String, String> lambdaIntegrationMap = new HashMap<String, String>();
    lambdaIntegrationMap.put("application/json", "{ \"statusCode\": \"200\" }");

    LambdaIntegration getWidgetIntegration =
        LambdaIntegration.Builder.create(lambdaFunction)
            .requestTemplates(lambdaIntegrationMap)
            .build();

    api.getRoot().addMethod("GET", getWidgetIntegration);

    LambdaIntegration postWidgetIntegration = new LambdaIntegration(lambdaFunction);
    LambdaIntegration deleteWidgetIntegration = new LambdaIntegration(lambdaFunction);

    Resource widget = api.getRoot().addResource("{id}");

    widget.addMethod("POST", postWidgetIntegration);
    widget.addMethod("GET", getWidgetIntegration);
    widget.addMethod("DELETE", deleteWidgetIntegration);
  }
}
